src/components/Header/Header.tsx

```diff
@@ -0,0 +1,96 @@
+import classNames from 'classnames';
+
+import { useEffect, useRef } from 'react';
+import { Todo } from '../../types/Todo';
+import { ErrorText } from '../../types/ErrorText';
+
+type Props = {
+  title: string,
+  onSetTitle: (newTitle: string) => void,
+  todos: Todo[],
+  onAddTodos: () => void,
+  onSetErrorMessage: (text: ErrorText) => void,
+  isDisabledInput: boolean,
+  onUpdateTodos: (newTodos: Todo[]) => void,
+  onIsFocusedHeader: boolean,
+};
+
+function getChangeTodosCompleted(todos: Todo[]) {
+  const changeTodosCompleted = todos.map(todo => {
+    return { ...todo, completed: !todo.completed };
+  });
+
+  return changeTodosCompleted;
+}
+
+export const Header: React.FC<Props> = ({
+  title,
+  onSetTitle,
+  todos,
+  onAddTodos,
+  onSetErrorMessage,
+  isDisabledInput,
+  onUpdateTodos,
+  onIsFocusedHeader,
+}) => {
+  const activeTodos = todos.filter(t => t.completed === false);
+
+  const handlerCompletedAll = () => {
+    const newTodos = (activeTodos.length < 1)
+      ? getChangeTodosCompleted(todos)
+      : getChangeTodosCompleted(activeTodos);
+
+    return onUpdateTodos(newTodos);
+  };
+
+  const handlerCreatedTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    onSetTitle(value);
+  };
+
+  const handlerSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (title.length < 1) {
+      onSetErrorMessage(ErrorText.EmptyTitle);
+
+      return;
+    }
+
+    onAddTodos();
+  };
+
+  const editedCompletedInput = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (editedCompletedInput.current !== null && onIsFocusedHeader) {
+      editedCompletedInput.current.focus();
+    }
+  }, [todos]);
```

Comment:
The dependency array in the 'useEffect' hook should also include 'onIsFocusedHeader' since it's being used in the hook.